airbyte/cloud/connectors.py

```diff
@@ -84,6 +87,116 @@ def __repr__(self) -> str:
         )
 
 
+class CloudConnectorVersionInfo(BaseModel):
+    """Information about a cloud connector's version."""
+
+    version: str
+    """The version string (e.g., '0.1.0')."""
+
+    is_version_pinned: bool
+    """Whether a version override is active for this connector."""
+
+    def __str__(self) -> str:
+        """Return a string representation of the version."""
+        return self.version if not self.is_version_pinned else f"{self.version} (pinned)"
+
+
+def _validate_version_override_permission(
+    *,
+    current_version: str,
+    new_version: str | None,
+    is_setting_override: bool,
+    override_reason: str,
+    user_email: str,
+    existing_override_creator_email: str | None = None,
+) -> tuple[bool, str]:
```

Comment:
Let's move this into a new submodule: `airbyte._util.biz_logic`. That is private already by being under _util, and it can be a public helper function (no prefix) from that module.